instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -192,7 +192,7 @@ private void setUrl(AttributeSetter setter, REQUEST request) {
       URI url = url(request);
       if (url != null) {
         netPeerAttributes.setNetPeer(setter, url.getHost(), null, url.getPort());
-        final URI sanitized;
+        URI sanitized;
```

Comment:
Why this is bad? If I want to guarantee that local variable is read-only, why is it bad to do so?

Reply:
@iNikem Do you mean specifically about this case where the variable is declared separately from initialization or any usage of `final` on local variable?

Reply:
Any usage of `final` on local variable. I agree that our codebase probably overdo it, but I don't like the idea of forbidding it.

Reply:
We decided against final on local variables a long time ago

#592

I think many of us agree with errorprone's sentiment that "it adds a fair amount of noise for minimal benefit."

https://errorprone.info/bugpattern/UnnecessaryFinal

I think it's worth being consistent on this and this prevents needing a yearly version of #592 going forward

Reply:
:( 😿

Reply:
we had carved out an exception for this particular case though:

> Local variables should only be declared final if they are not initialized inline (declaring these vars final can help prevent accidental double-initialization).

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/contributing/style-guideline.md#final-keyword-usage

While I think it's a nice exception, if errorprone can't differentiate, I'm ok with removing the exception.

Reply:
Ah forgot we documented it. Indeed I lean towards sticking with the error prone check to guarantee consistency instead.

Through this I learned that error prone actually tries to only use syntax in the rarer case - there is a `Var` check which will fail if the variable is assigned to multiple times, what this final was intended to avoid. For legitimate cases you add a `@Var` errorprone annotation. It's not clear to me yet if it's worth bringing in errorprone annotations just for this but could be a good approach - currently we have the Var check disabled.